packages/web/src/shared/components/form/ReorderableMultiselectInput.tsx

```diff
@@ -0,0 +1,152 @@
+import { forwardRef, useState } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { Icon } from '../Icon';
+import Button from '../actions/Button';
+import ComboboxInput from './ComboboxInput';
+
+export type ReorderableMultiselectInputProps =
+  BaseReorderableMultiselectInputProps & {
+    required?: boolean;
+    isolate?: boolean;
+  };
+
+export default function ReorderableMultiselectInput(
+  props: ReorderableMultiselectInputProps
+) {
+  const context = useFormContext();
+
+  if (context && !props.isolate) {
+    return (
+      <Controller
+        control={context.control}
+        name={props.name}
+        defaultValue={props.defaultValue}
+        rules={{ required: props.required }}
+        render={({ field, fieldState }) => (
+          <BaseReorderableMultiselectInput
+            {...field}
+            items={props.items}
+            hasErrors={!!fieldState.error}
+            placeholder={props.placeholder}
+          />
+        )}
+      />
+    );
+  } else {
+    return <BaseReorderableMultiselectInput {...props} />;
+  }
+}
+
+interface BaseReorderableMultiselectInputProps {
+  className?: string;
+  name: string;
+  items: ItemType[];
+  value?: string[];
+  defaultValue?: string[];
+  hasErrors?: boolean;
+  placeholder?: string;
+  onChange?(value: string[]): void;
+  onBlur?(): void;
+}
+
+type ItemType = { label: string; value: string };
+
+const BaseReorderableMultiselectInput = forwardRef<
+  HTMLInputElement,
+  BaseReorderableMultiselectInputProps
+>(({ className = '', value, onChange, items }, ref) => {
```

Comment:
We need to add onBlur here so that react-hook-form can tell when this input has been touched. Maybe we can call onBlur when any child element inside the component blurs

Reply:
Done. Will this cause issues when the user tabs between the different children elements? `onBlur` will be called multiple times, while the entire component still has focus.